File: dart/glue/lib/src/lib/bool/empty.dart
import 'package:glue/src/eval.dart';
import 'package:glue/src/ir.dart';
import 'package:glue/src/ir.dart' as ir;

final Ir empty = IrNativeFunc(emptyImpl);

Eval<Ir> emptyImpl(Ir arg) => Eval.pure(IrBool(ir.empty(arg)));

File: haskell/glue/src/Glue/Lib/Bool/Empty.hs
module Glue.Lib.Bool.Empty where

import Glue.Eval (Eval)
import Glue.IR as IR (IR (..), empty)

empty :: IR Eval
empty = NativeFunc emptyImpl

emptyImpl :: IR Eval -> Eval (IR Eval)
emptyImpl = pure . Bool . IR.empty
